src/principal-vs-secondary/regis/Diagonal.java

```diff
@@ -0,0 +1,66 @@
+import java.util.Random;
+public class Diagonal {
+
+    public int [][] matrixRandom(int side){
+        int [][] matrix = new int [side][side];
+        Random numRand =  new Random();
+        for (int x = 0; x < side; x++) {
+            for (int y = 0; y < side; y++) {
+                matrix[x][y] = numRand.nextInt(9) + 1;
+            }
+        }
+        return matrix;
+    }
+
+    public int addDiagonalMain(int [][] matrix){
+        int result = 0;
+        for (int x = 0; x < matrix.length; x++) {
+            for (int y = 0; y < matrix.length; y++) {
+                if(x == y){
+                    result = result + matrix[x][y];
+                }
+            }
+        }
+        System.out.println("Diagonal Main: " + result);
+        return result;
+    }
+
+    public int addDiagonalSecondary(int [][] matrix){
```

Comment:
Could this method and the previous one be one single method? maybe its difference could be handled by a single `if-then-else` and a parameter flag